src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php

```diff
@@ -1227,6 +1252,71 @@ public function findServiceGroupsByHostAndService(int $hostId, int $serviceId):
         return $serviceGroups;
     }
 
+    /**
+     * Find downtimes for host or service
+     * @param int $hostId
+     * @param int $serviceId
+     * @return array
+     */
+    public function findDowntimes(int $hostId, int $serviceId): array
+    {
+        $downtimes = [];
+
+        if (empty($hostId)) {
+            return $downtimes;
+        }
+
+        $sql = 'SELECT * FROM `:dbstg`.`downtimes` WHERE host_id = :hostId AND service_id = :serviceId ' .
+                'AND deletion_time IS NULL ORDER BY entry_time DESC LIMIT 500';
+        $request = $this->translateDbName($sql);
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':hostId', $hostId, \PDO::PARAM_INT);
+        $statement->bindValue(':serviceId', $serviceId, \PDO::PARAM_INT);
+        $statement->execute();
+
+        while ($result = $statement->fetch(\PDO::FETCH_ASSOC)) {
+            $downtimes[] = EntityCreator::createEntityByArray(
+                Downtime::class,
+                $result
+            );
+        }
+
+        return $downtimes;
+    }
+
+    /**
+     * Find acknowledgements for host or service
+     * @param int $hostId
+     * @param int $serviceId
+     * @return array
+     */
+    public function findAcknowledgements(int $hostId, int $serviceId): array
+    {
+        $acks = [];
+
+        if (empty($hostId)) {
+            return $acks;
+        }
+
+        $sql = 'SELECT * FROM `:dbstg`.`acknowledgements` WHERE host_id = :hostId AND service_id = :serviceId ' .
+                'AND deletion_time IS NULL ORDER BY entry_time DESC LIMIT 500';
```

Comment:
```suggestion
                'AND deletion_time IS NULL ORDER BY entry_time DESC LIMIT 1';
```

Reply:
This method actually fetches all acknowledgements. Further ahead in the logic I set only the first element of the returned for this particular use case. Tomorrow, we might want to list all of them for a host, expired ones also and we can reuse.

Reply:
there is already an endpoint for this : https://github.com/centreon/centreon/blob/MON-4789-unified-view/doc/API/centreon-api-v2.yaml#L770
Here, we just want current acknowledgement